flow360/component/project.py

```diff
@@ -0,0 +1,563 @@
+                length_unit
+            )
+
+        root_asset = None
+        root_type = self.metadata.root_item_type
+
+        if root_type == RootType.GEOMETRY:
+            root_asset = self._geometry
+        elif root_type == RootType.VOLUME_MESH:
+            root_asset = self._volume_mesh
+
+        if not root_asset:
+            raise Flow360ValueError("Root asset is not initialized")
+
+        draft = Draft.create(
+            name=draft_name,
+            project_id=self.metadata.id,
+            source_item_id=self.metadata.root_item_id,
+            source_item_type=root_type.value,
+            solver_version=self.solver_version,
+            fork_case=fork,
+        ).submit()
+
+        entity_info = root_asset.entity_info
+        registry = params.used_entity_registry
+        old_draft_entities = entity_info.draft_entities
+        for _, old_entity in enumerate(old_draft_entities):
+            try:
+                registry.find_by_naming_pattern(old_entity.name)
+            except ValueError:
+                continue
+
+        with model_attribute_unlock(params.private_attribute_asset_cache, "project_entity_info"):
+            params.private_attribute_asset_cache.project_entity_info = entity_info
+
+        draft.update_simulation_params(params)
+        destination_id = draft.run_up_to_target_asset(target)
+
+        RestApi(ProjectInterface.endpoint, id=self.metadata.id).patch(
+            json={
+                "lastOpenItemId": destination_id,
+                "lastOpenItemType": target.__name__,
+            }
+        )
+
+        destination_obj = target.from_cloud(destination_id)
+
+        if not run_async:
+            destination_obj.wait()
+
+        return destination_obj
+
+    def run_surface_mesher(
+        self,
+        params: SimulationParams,
+        draft_name: str = "SurfaceMesh",
+        run_async: bool = True,
+        fork: bool = False,
+    ):
+        """
+        Runs the surface mesher for the project.
+
+        Parameters
+        ----------
+        params : SimulationParams
+            Simulation parameters for running the mesher.
+        draft_name : str, optional
+            Name of the draft (default is "SurfaceMesh").
+        run_async : bool, optional
+            Whether to run the mesher asynchronously (default is True).
+        fork : bool, optional
+            Whether to fork the case (default is False).
+
+        Raises
+        ------
+        Flow360ValueError
+            If the root item type is not Geometry.
+        """
+        self._check_initialized()
+        if self.metadata.root_item_type is not RootType.GEOMETRY:
+            raise Flow360ValueError(
+                "Surface mesher can only be run by projects with a geometry root asset"
+            )
+        self._surface_mesh = self._run(
+            params=params, target=SurfaceMesh, draft_name=draft_name, run_async=run_async, fork=fork
+        )
+
+    def run_volume_mesher(
```

Comment:
```suggestion
    def generate_volume_mesh(
```

Reply:
Fixed in [9ba5587](https://github.com/flexcompute/Flow360/pull/511/commits/9ba5587bee8a37191788f1094fe8503dd5cda41d)